sklearn/compose/tests/test_column_transformer.py

```diff
@@ -649,25 +657,53 @@ def test_column_transformer_get_feature_names():
     ct.fit(X)
     assert_equal(ct.get_feature_names(), ['col0__a', 'col0__b', 'col1__c'])
 
-    # passthrough transformers not supported
+    # drop transformer
+    ct = ColumnTransformer(
+        [('col0', DictVectorizer(), 0), ('col1', 'drop', 1)])
+    ct.fit(X)
+    assert_equal(ct.get_feature_names(), ['col0__a', 'col0__b'])
+
+    # passthrough transformer
     ct = ColumnTransformer([('trans', 'passthrough', [0, 1])])
     ct.fit(X)
-    assert_raise_message(
-        NotImplementedError, 'get_feature_names is not yet supported',
-        ct.get_feature_names)
+    assert_equal(ct.get_feature_names(), ['trans__0', 'trans__1'])
```

Comment:
I'm not convinced that we should be applying a prefix if trans does nothing... Can you imagine a context where this helps? Yes, it avoids naming conflicts...

Reply:
This was just to keep it consistent with the 'name__column' syntax used for when an explicit name is passed with a transformer, if for any reason someone passed transformers=[(name, 'passthrough', [columns])]. This seemed more consistent than dropping the name.

Reply:
Hmmm.... I still am not happy with this prefixing. I'd be interested to hear other opinions, or use-cases.

Reply:
I'm not sure why 'passthrough' is a valid option for a transformer at all, it seems like that could just be handled with remainder='passthrough' and dropping any unused columns before using the ColumnTransformer. But as it can be passed as a transformer with a name, it seems to make sense to use that name on the feature names, like with the other transformers. Having said that, these should probably be 'trans_x0', 'trans_x1' to be consistent with that notation. Happy to change it altogether though if there is a better option.

Reply:
> I'm not sure why 'passthrough' is a valid option for a transformer at all, it seems like that could just be handled with remainder='passthrough' and dropping any unused columns before using the ColumnTransformer.

It is available to allow the user to try disabling some transformations in a parameter search such as grid search.

Reply:
Ah okay, so I suppose in that case you would want those disabled columns to be treated the same way as all of the other passthrough columns. I will make this change.